daos/book.js

```diff
@@ -4,8 +4,73 @@ const Book = require('../models/book');
 
 module.exports = {};
 
-module.exports.getAll = (page, perPage) => {
-  return Book.find().limit(perPage).skip(perPage*page).lean();
+module.exports.getAll = (page, perPage, authorId) => {
+  //return Book.find().limit(perPage).skip(perPage*page).lean();
+
+  //Include filter for authorId if specified and modified to use aggregate query
+  let findFilter = authorId ? { authorId : authorId } : {}
```

Comment:
```suggestion
  const findFilter = authorId ? { authorId } : {}
```
curious if ES6 would work here